spring-cloud-kubernetes-fabric8-config/src/main/java/org/springframework/cloud/kubernetes/fabric8/config/LabeledSecretContextToSourceDataProvider.java

```diff
@@ -77,9 +81,43 @@ public Fabric8ContextToSourceData get() {
 						.getItems();
 
 				if (!secrets.isEmpty()) {
-					secrets.forEach(secret -> result.putAll(dataFromSecret(secret, namespace)));
-					sourceName = secrets.stream().map(Secret::getMetadata).map(ObjectMeta::getName)
+
+					for (Secret secret : secrets) {
+						// we support prefix per source, not per secret. This means that
```

Comment:
when we search "by labels" - we can find more than one secret, these secrets we find can easily have the same property names - and they will get overridden. This is expected and more fine-grained labels would be the solution here.